### DISCUSSION TITLE
Stress-strain curve

### DISCUSSION THREAD

**Discussion by xchengood** [idaholab/moose#24113] [Q&A Modules: Solid mechanics] [answered]
Hello,

I am studying _CrystalPlasticityHCPDislocationSlipBeyerleinUpdate_ class and performing example files like _update_method_hcp_no_substructure.i_. The question is that there is only an elastic part and no yield occurs in the stress-strain curve below.  What parameters should I change?
<img width="350" alt="image" src="https://user-images.githubusercontent.com/74226455/232840603-4238fd01-df1e-43e5-b4b1-ff8f0c0a2fca.png">
When I change _end_time_ from 2.5 to 250, I got the error below. Any ideas about this?
![image](https://user-images.githubusercontent.com/74226455/232841109-213c2a57-3cf4-4a40-a7e8-5e48c9a55166.png)

Thank you.

**Comment by sapitts** [comment, marked as answer]
Hi @xchengood, 
This particular input file is designed to artificially test a corner case in the code, and I wouldn't expect this set of input parameters to produce realistic results, see the comment in the input file here:
https://github.com/idaholab/moose/blob/d295aad6bd1e6f228b67b2888a4cd6b787d1f659/modules/tensor_mechanics/test/tests/crystal_plasticity/hcp_single_crystal/update_method_hcp_no_substructure.i#L127

To help guide your efforts in finding realistic vs code regression testing input files, you can look at the `requirement` and `detail` lines in the [tests](https://github.com/idaholab/moose/blob/next/modules/tensor_mechanics/test/tests/crystal_plasticity/hcp_single_crystal/tests) file. And the published papers cited in the [class documentation](https://mooseframework.inl.gov/source/materials/crystal_plasticity/CrystalPlasticityHCPDislocationSlipBeyerleinUpdate.html)

Hope this helps,
Stephanie
